(Mingw) Citra no longer compiles with SDL2 disabled

Bisected to ab021d163e2a33b25b1811dc6f6abd924ec2d529 @lioncash 

Citra no longer compiles with ENABLE_SDL2 set to 0 on, at least, Mingw platforms.

CMake generation log: https://gist.github.com/j-selby/a61bfb10d544952a095d17cefcb2f50e
tl;dr:
`-DCMAKE_BUILD_TYPE=RelWithDebInfo -DENABLE_WEB_SERVICE=0 -DENABLE_QT=1 -DENABLE_SDL2=0 -G "CodeBlocks - MinGW Makefiles"`

CMake build log: https://gist.github.com/j-selby/3e9f9437e3c842bcb889b751a702f79f
tl;dr:
```
[ 31%] Building CXX object src/audio_core/CMakeFiles/audio_core.dir/sdl2_sink.cpp.obj
<path-to>\citra\src\audio_core\sdl2_sink.cpp:7:10: fatal error: SDL.h: No such file or directory
 #include <SDL.h>
          ^~~~~~~
compilation terminated.
```
Tested commit: `c143a5a3`
CMake version: `3.9.1`
Platform: `Windows 10.0.16299.125`

--- a/src/audio_core/CMakeLists.txt
+++ b/src/audio_core/CMakeLists.txt
@@ -23,7 +23,7 @@ add_library(audio_core STATIC
     time_stretch.cpp
     time_stretch.h
 
-    $<$<BOOL:SDL2_FOUND>:sdl2_sink.cpp sdl2_sink.h>
+    $<$<BOOL:${SDL2_FOUND}>:sdl2_sink.cpp sdl2_sink.h>
 )
 
 create_target_directory_groups(audio_core)

--- a/src/input_common/CMakeLists.txt
+++ b/src/input_common/CMakeLists.txt
@@ -8,7 +8,7 @@ add_library(input_common STATIC
     motion_emu.cpp
     motion_emu.h
 
-    $<$<BOOL:SDL2_FOUND>:sdl/sdl.cpp sdl/sdl.h>
+    $<$<BOOL:${SDL2_FOUND}>:sdl/sdl.cpp sdl/sdl.h>
 )
 
 create_target_directory_groups(input_common)

